apps/src/JSInterpreter.js

```diff
@@ -423,8 +433,10 @@ JSInterpreter.prototype.executeInterpreter = function (firstStep, runUntilCallba
     }
     var err = safeStepInterpreter(this);
     if (!err) {
-      doneUserLine = doneUserLine ||
-        (inUserCode && this.interpreter.stateStack[0] && this.interpreter.stateStack[0].done);
+      var nodeType = this.interpreter.stateStack[0].node.type;
+      if (inUserCode) {
+        doneUserLine = doneUserLine || INTERSTITIAL_NODES.hasOwnProperty(nodeType);
```

Comment:
Style nit: any reason not to write `INTERSTITIAL_NODES[nodeType]`? It looks cleaner to me :smile: